Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -89,6 +90,30 @@ VerifierOptions verifierOptions
       );
     }
 
+    public async Task<DafnyDocument> LoadAndPrepareVerificationTasksAsync(TextDocumentItem textDocument, CancellationToken cancellationToken) {
+      var loaded = await LoadAsync(textDocument, cancellationToken);
+      if (loaded.ParseAndResolutionDiagnostics.Any(d => d.Severity == DiagnosticSeverity.Error)) {
+        return loaded;
+      }
+
+      var verificationView = verifier.GetVerificationTasks(loaded);
```

Comment:
Good point. I'll make sure that resolution diagnostics are returned as soon as resolution is done, and that we don't start resolution or verification task computation work in case this is outdated.